### PR TITLE
Add JPEG Quality Option

### PR DESCRIPTION
**Pull request by mdphillips375**
JPEG is a lossy compression algorithm that trades quality for file size.  This update adds an option to the general options which allows the user to set the quality and passes that on in the places where there is a possibility to write an image (either to a file or buffer) in JPEG format.  The changes to the calls to QPixmap::save() and QImageWriter::setQuality() are wrapped in if statements to check the format so that these changes do not impact (the probably more common) PNG format, which also uses the same parameter differently (a PNG with maximum compression is still lossless).

This change addresses issue #3216 and #506 by giving the user the choice of a higher quality, but larger file.

### CODE REVIEW

File: src/config/generalconf.cpp
Comment (mmahmoudian) [inline src/config/generalconf.cpp:785]: Is the "timeout" variable name intentional? If not, I suggest changing it to something more fitting to improve code readability

**Comment by mmahmoudian** [comment]
@mdphillips375 Thanks for the PR and the time you have invested in Flameshot. Let's wait for other devs to also go through the code and approve it. Cheers.

File: src/config/generalconf.cpp
Comment (mdphillips375) [reply src/config/generalconf.cpp]: The fix has been pushed.

**Comment by jack9603301** [comment]
Looks good

**Comment by jack9603301** [comment]
图片

I think you can clean up those commits, they are pointless for this PR

**Comment by mmahmoudian** [comment]
@mdphillips375 thanks for handling the comments. I set it to run the Actions.

One question though: the range is 0-100. What does 0 and 100 mean? Is 100 without any compression? And what is 0?

**Comment by mdphillips375** [comment]
Looking through the sources to libjpeg, I see several references to 100 being lossless.  It certainly has the visual appearance of being lossless.  I can say that 0 is pretty bad for most things I would personally use this software for.  A screenshot of text is not legible.  It will turn a photograph into an image resembling a low resolution 8-bit (or lower) display, which could actually be desirable in some cases (retro gaming look).

Qt5 defaults to setting the quality to 75 if you do not pass a quality parameter, which is the case in the current release, so I set the default to 75.

As for a size comparison of the 100 setting, I took a screenshot test I did with Quality = 100, converted to uncompressed TIF, and PNG with maximum compression using GIMP.  It is certainly not uncompressed, although not as small as PNG's lossless compression.

Size Comparison of screens

**Comment by mmahmoudian** [comment]
@panpuchkov @veracioux do you see any issues to be discussed here or do you have any objections?

**Review by veracioux — CHANGES_REQUESTED** [review]
Please also merge `master` into this branch, so that this PR can be tested against a CI/CD pipeline that has had some fixes applied.

File: src/utils/confighandler.h
Comment (mdphillips375) [reply src/utils/confighandler.h]: Done

File: src/config/generalconf.cpp
Comment (mdphillips375) [reply src/config/generalconf.cpp]: Done.  Was wondering why that was not used, but went with consistency with the surrounding code.

File: src/utils/confighandler.cpp
Comment (mdphillips375) [reply src/utils/confighandler.cpp:128]: Done

**Comment by veracioux** [comment]
@mdphillips375 Thanks for the changes. Please also run clang-format as described [here](https://github.com/flameshot-org/flameshot/blob/master/docs/CONTRIBUTING.md#code). We require this on all PRs before merge.

**Comment by mdphillips375** [comment]
Fixed.  Sorry about that.

**Pull request merged**
